### PR TITLE
Update schema to use new definitions groups

### PR DESCRIPTION
**Pull request by ewels**
New pipeline schema structure. See nf-core/tools#696

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -90,7 +92,8 @@
             },
             "fa_icon": "fas fa-cloud-download-alt"
         },
-        "Input Data Additional Options": {
+        "input_data_additional_options": {
+            "title": "Input Data Additional Options",
```
Comment (jfy133) [inline nextflow_schema.json:96]:
```suggestion
            "title": "Input data additional options",
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -382,7 +391,8 @@
             },
             "fa_icon": "fas fa-sort-amount-down"
         },
-        "Read Clipping and Adapter Removal": {
+        "read_clipping_and_adapter_removal": {
+            "title": "Read Clipping and Adapter Removal",
```
Comment (jfy133) [inline nextflow_schema.json:395]:
```suggestion
            "title": "Read merging and adapter removal",
```

File: nextflow_schema.json [hunk 3]
```diff
@@ -382,7 +391,8 @@
             },
             "fa_icon": "fas fa-sort-amount-down"
         },
-        "Read Clipping and Adapter Removal": {
+        "read_clipping_and_adapter_removal": {
```
Comment (jfy133) [inline nextflow_schema.json:394]:
```suggestion
        "read_merging_and_adapter_removal": {
```

File: nextflow_schema.json [hunk 4]
```diff
@@ -1082,5 +1108,85 @@
...
+            "$ref": "#/definitions/institutional_config_options"
+        },
+        {
+            "$ref": "#/definitions/skip_steps"
+        },
+        {
+            "$ref": "#/definitions/complexity_filtering"
+        },
+        {
+            "$ref": "#/definitions/read_clipping_and_adapter_removal"
```
Comment (jfy133) [inline nextflow_schema.json:1141]:
```suggestion
            "$ref": "#/definitions/read_merging_and_adapter_removal"
```

**Comment by ewels** [comment]
Go for it 👍🏻

**Pull request merged**